// @shihengtech/hooks/lib/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "useEventCallback", {
  enumerable: true,
  get: function get() {
    return _useEventCallback.default;
  }
});
Object.defineProperty(exports, "useFetch", {
  enumerable: true,
  get: function get() {
    return _useFetch.default;
  }
});
Object.defineProperty(exports, "useForceUpdate", {
  enumerable: true,
  get: function get() {
    return _useForceUpdate.default;
  }
});
Object.defineProperty(exports, "useImmutable", {
  enumerable: true,
  get: function get() {
    return _useImmutable.default;
  }
});
Object.defineProperty(exports, "useImmutableFunc", {
  enumerable: true,
  get: function get() {
    return _useImmutableFunc.default;
  }
});
Object.defineProperty(exports, "useLoading", {
  enumerable: true,
  get: function get() {
    return _useLoading.default;
  }
});
Object.defineProperty(exports, "useLocalStorageState", {
  enumerable: true,
  get: function get() {
    return _useLocalStorageState.default;
  }
});
Object.defineProperty(exports, "useModal", {
  enumerable: true,
  get: function get() {
    return _useModal.default;
  }
});
Object.defineProperty(exports, "usePrevious", {
  enumerable: true,
  get: function get() {
    return _usePrevious.default;
  }
});
Object.defineProperty(exports, "useSelections", {
  enumerable: true,
  get: function get() {
    return _useSelections.default;
  }
});
Object.defineProperty(exports, "useToggle", {
  enumerable: true,
  get: function get() {
    return _useToggle.default;
  }
});
Object.defineProperty(exports, "useUpdateEffect", {
  enumerable: true,
  get: function get() {
    return _useUpdateEffect.default;
  }
});

var _useModal = _interopRequireDefault(require("./useModal"));

var _useSelections = _interopRequireDefault(require("./useSelections"));

var _useLocalStorageState = _interopRequireDefault(require("./useLocalStorageState"));

var _useLoading = _interopRequireDefault(require("./useLoading"));

var _useEventCallback = _interopRequireDefault(require("./useEventCallback"));

var _useFetch = _interopRequireDefault(require("./useFetch"));

var _useImmutable = _interopRequireDefault(require("./useImmutable"));

var _useImmutableFunc = _interopRequireDefault(require("./useImmutableFunc"));

var _useUpdateEffect = _interopRequireDefault(require("./useUpdateEffect"));

var _usePrevious = _interopRequireDefault(require("./usePrevious"));

var _useForceUpdate = _interopRequireDefault(require("./useForceUpdate"));

var _useToggle = _interopRequireDefault(require("./useToggle"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
